### DISCUSSION TITLE
Allow for environment variables to set settings

### DISCUSSION THREAD

**Discussion by ngriffiths13** [marimo-team/marimo#1370] [Ideas]
It would be really convenient to allow for the use of environment variables to define settings. Specifically I think this would be useful when deploying via Docker. I don't really care about most of the marimo settings when deploying an app because I am no longer developing, but would love to easily be able to do something like:

`ENV MARIMO_THEME="dark"`

**Comment by mscolnick** [comment]
This is something we want to do: bridge CLI args, .marimo.toml, and env variables into one resolved config. we just havent gotten to it yet

**Reply by ngriffiths13** [reply]
Maybe I'll take a look at it. Do you think this would be pretty straightforward?

**Reply by mscolnick** [reply]
I think it would involve coming up with a consistent / neat design that almost makes new config automatically work across all 3.
i think though we just start with 2 (env and marimo.toml). 

we have the `UserConfigManager` which could handle all this. just thinking out-loud, but could look something like:

```
- abstract class MarimoConfigGetter
- TomlMarimoConfigGetter
- EnvMarimoConfigGetter
- DefaultMarimoConfigGetter
- CompositeMarimoConfigGetter([EnvMarimoConfigGetter, TomlMarimoConfigGetter, DefaultMarimoConfigGetter])
```
